### PR TITLE
Cleanup code

### PR DESCRIPTION
**Pull request by andreyrcdias**
### Problem
No problem just simplified the code. These changes do not impact any project functionality.

### Solution
Code more tiny and pythonic.

### Notes
Am I forgetting something? Should I create the Issue?

### CODE REVIEW

File: lib/charms/mongodb/v0/mongodb.py [hunk 1]
```diff
@@ -414,10 +414,7 @@ def _is_any_sync(rs_status: Dict) -> bool:
             rs_status: current state of replica set as reported by mongod.
         """
         return any(
-            member["stateStr"] == "STARTUP"
```
Comment (MiaAltieri) [inline lib/charms/mongodb/v0/mongodb.py:417]: Hi @andreyrcdias thanks for helping us improve our code :)

The code in `lib/charms/mongodb` are part of our [charm libraries](https://juju.is/docs/sdk/libraries). When changes are made in the libary it is important to bump the `LIBPATCH` constant at the top of the file.

Definition of LIBPATCH:
> Current patch version of the library. Must be updated each time a new version of the library is published to Charmhub.

File: lib/charms/mongodb/v0/mongodb_backups.py [hunk 2]
```diff
@@ -177,9 +177,7 @@ def _resync_config_options(self, pbm_snap):
         ):
             with attempt:
                 pbm_status = self._get_pbm_status()
-                if isinstance(pbm_status, MaintenanceStatus) or isinstance(
-                    pbm_status, WaitingStatus
-                ):
+                if isinstance(pbm_status, (MaintenanceStatus, WaitingStatus)):
```
Comment (MiaAltieri) [inline lib/charms/mongodb/v0/mongodb_backups.py:180]: Hi @andreyrcdias thanks for helping us improve our code :)

The code in `lib/charms/mongodb` are part of our [charm libraries](https://juju.is/docs/sdk/libraries). When changes are made in the libary it is important to bump the `LIBPATCH` constant at the top of the file.

Definition of LIBPATCH:
> Current patch version of the library. Must be updated each time a new version of the library is published to Charmhub.

**Review by MiaAltieri — CHANGES_REQUESTED** [review]
Awesome contributions here @andreyrcdias we are appreciative of these improvements :) The library versions need to get updated so that the [libary check passes](https://github.com/canonical/mongodb-operator/actions/runs/4486428515/jobs/7890598002?pr=190)

**Review by delgod — APPROVED** [review]

File: lib/charms/mongodb/v0/mongodb.py (hunk 1 above)
Comment (andreyrcdias) [reply lib/charms/mongodb/v0/mongodb.py:417]: okay, all right. Thanks

**Comment by andreyrcdias** [comment]
> Awesome contributions here @andreyrcdias we are appreciative of these improvements :) The library versions need to get updated so that the [libary check passes](https://github.com/canonical/mongodb-operator/actions/runs/4486428515/jobs/7890598002?pr=190)

Would it just be incrementing the LIBPATH version?

**Comment by MiaAltieri** [comment]
@andreyrcdias mind merging main into your branch #189 just merged and it has fixes for some tests that are failing along with updated decencies :)

**Comment by MiaAltieri** [comment]
Hi @andreyrcdias sorry for the delays here. Currently working on a work-around to get the tests to run 

The test is failing because the action `check-lib` need a credentials to fetch libraries. It has been improved already here (but not released): https://github.com/canonical/charmcraft/pull/1037

Since your PR is an external PR it has no access to the secret stored in GitHub.

This will be fixed automatically as soon as new charmcraft is released. But until then I will find another way to run these tests. Thank you for your patience

**Comment by MiaAltieri** [comment]
Update @andreyrcdias -

I've copied your changes to my own branch and ran the tests [they pass](https://github.com/canonical/mongodb-operator/pull/193) I will accept your changes and merge.

Thanks for your contributions here and I look forward to seeing whats next! :D

**Review by MiaAltieri — APPROVED** [review]
awesome work! Thank you :)

**Pull request merged**